app/actions/remote/channel.ts

```diff
@@ -404,10 +404,74 @@ export async function fetchChannelStats(serverUrl: string, channelId: string, fe
     }
 }
 
+export async function fetchAllMyChannelsForAllTeams(serverUrl: string, since: number, isCRTEnabled: boolean, fetchOnly = false, groupLabel?: RequestGroupLabel) {
+    try {
+        if (!fetchOnly) {
+            setTeamLoading(serverUrl, true);
+        }
+        const client = NetworkManager.getClient(serverUrl);
+        const {operator} = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+        const promises: [Promise<Channel[]>, Promise<ChannelMembership[]>] = [
+            client.getAllChannelsFromAllTeams(since, since > 0, groupLabel),
+            client.getAllMyChannelMembersFromAllTeams(0, General.CHANNEL_MEMBERS_CHUNK_SIZE, groupLabel),
+        ];
+
+        const [channels, memberships] = await Promise.all(promises);
+        const remaining = Math.floor(channels.length / General.CHANNEL_MEMBERS_CHUNK_SIZE);
+        const remainingMembershipsPromises = [];
+        const categoriesPromises = [];
+
+        const teamIdsSet = channels.reduce<Set<string>>((set, c) => {
+            if (c.team_id) {
+                set.add(c.team_id);
+            }
+            return set;
+        }, new Set());
+        for (const teamId of teamIdsSet) {
+            categoriesPromises.push(client.getCategories('me', teamId, groupLabel));
+        }
```

Comment:
If we have an absurd amount of teams, we are flooding the network layer with get categories calls. It is not common, but we have already had reports of people having 100+ teams.

Unless we create a priority queue down at the network layer, we may want to consider batching this, so other processes can access the network layer before this finishes.

Reply:
Agreed.  Thanks for your observation.  This is something this PR/changes is trying to establish on where the bottlenecks are and learn from it. And prioritize all the issues.  We have a long way to go, but first, data.